dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -140,15 +142,33 @@ bool TarArchive::NextFile() {
   if (eof_) {
     return false;
   }
-  Skip(RoundToBlockSize(filesize_) - readoffset_);
-  eof_ = ParseHeader();
+
+  if ((archiveoffset_ += RoundToBlockSize(filesize_) - readoffset_) >= stream_->Size()) {
+    SetEof();
+    return false;
+  }
+  stream_->Seek(archiveoffset_);
+
+  ParseHeader();
   return !eof_;
 }
 
 bool TarArchive::EndOfArchive() const {
   return eof_;
 }
 
+void TarArchive::Seek(size_t offset) {
+  DALI_ENFORCE(offset % T_BLOCKSIZE == 0, "Invalid tar archive offset");
```

Comment:
How about having an `assert` here? With `DALI_ENFORCE` I'm afraid, that when the exception is thrown, the user is prompted with `"Invalid tar archive offset"` during training. And this doesn't tell him much...

Reply:
done